tests/console/docker.pm

```diff
@@ -33,72 +37,57 @@ sub run {
         die "Docker is not pre-installed." if script_run("zypper se -x --provides -i docker | grep docker");
     }
     else {
-        zypper_call("in docker");
+        zypper_call("in docker"); # docker package can be installed
     }
 
-    # start the docker daemon
+    # docker daemon can be started
     systemctl("start docker");
-
-    # check status of docker daemon
     systemctl("status docker");
+    assert_script_run('docker info');
 
-    # do search for openSUSE
+    # images can be searched on the Docker Hub
     validate_script_output("docker search --no-trunc opensuse", sub { m/This project contains the stable releases of the openSUSE distribution/ });
 
-    assert_script_run('docker info');
-
+    # images can be pulled from the Docker Hub
+    #   - pull minimalistic alpine image of declared version
+    #   - https://store.docker.com/images/alpine
     my $alpine_image_version = '3.5';
-    # pull minimalistic alpine image of declared version
-    # https://store.docker.com/images/alpine
     assert_script_run("docker pull alpine:$alpine_image_version", 300);
-
-    # Check if the alpine image has been fetched
     assert_script_run("docker images -q alpine:$alpine_image_version");
-
-    # pull hello-world image, typical docker demo image
-    # https://store.docker.com/images/hello-world
+    #   - pull hello-world image, typical docker demo image
+    #   - https://store.docker.com/images/hello-world
     assert_script_run("docker pull hello-world", 300);
-
-    # Check if the hello-world image has been fetched
     assert_script_run("docker images -q hello-world:latest");
+    
+    # local images can be listed
+    validate_script_output("docker images", sub { m/alpine/ });
```

Comment:
Do you like that more than `assert_script_run 'docker images | grep alpine'`? I think this is more readable, more efficient and easier to understand from screenshots. The `validate_script_output` calls tend to confuse people because it produces a lot of "curl" output hiding the actual important calls. The call to assert_script_run is shown in screenshots as is with just a little bit of extra for appending the exit code magic string to the serial port

Reply:
I haven written that instruction, with this PR I only wanted to reorganized the existing instructions and improve comments.

My plan was after this PR would be eventually accepted, create following PR to improve the test. (for example some commands will give false negatives...) What you suggest, I reserved for following PR's